dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/SessionFactoryHealthCheck.java

```diff
@@ -49,7 +50,7 @@ protected Result check() throws Exception {
                     session.createSQLQuery(validationQuery).list();
                     txn.commit();
                 } catch (Exception e) {
-                    if (txn.isActive()) {
+                    if (txn.getStatus() == ACTIVE || txn.getStatus() == MARKED_ROLLBACK) {
```

Comment:
Why not use [`TransactionStatus#canRollback()`](https://docs.jboss.org/hibernate/orm/5.0/javadocs/org/hibernate/resource/transaction/spi/TransactionStatus.html#canRollback--) here?